lizard_auth_server/models.py

```diff
@@ -746,3 +740,136 @@ def as_dict(self):
             "organisation": self.organisation.as_dict(),
             "role": self.role.as_dict()
         }
+
+
+# New models
+# ##########
+
+class Profile(models.Model):
+    """Replacement for UserProfile."""
+    user = models.OneToOneField(
+        User,
+        verbose_name=_('user'),
+        related_name='profile')
+    company = models.ForeignKey(
+        'Company',
+        verbose_name=_('company'),
+        blank=True,
+        null=True)
+    created_at = models.DateTimeField(
+        verbose_name=_('created on'),
+        auto_now_add=True)
+    updated_at = models.DateTimeField(
+        verbose_name=_('updated on'),
+        auto_now=True)
+
+    def __str__(self):
+        return self.user.username
+
+    class Meta:
+        verbose_name = _("user profile")
+        verbose_name_plural = _("user profiles")
+
+    @property
+    def username(self):
+        return self.user.username
+
+    @property
+    def full_name(self):
+        return self.user.get_full_name()
+
+    @property
+    def first_name(self):
+        return self.user.first_name
+
+    @property
+    def last_name(self):
+        return self.user.last_name
+
+    @property
+    def email(self):
+        return self.user.email
+
+
+class Company(models.Model):
+    """Replacement for Organisation."""
+    name = models.CharField(
+        verbose_name=_('name'),
+        max_length=255,
+        null=False,
+        blank=False,
+        unique=True)
+    unique_id = models.CharField(
+        verbose_name=_('unique id'),
+        max_length=32,
+        unique=True,
+        default=create_new_uuid)
+    guests = models.ManyToManyField(
+        'Profile',
+        related_name='guest_companies',
```

Comment:
De naam vind ik niet 100% lekker. "guest" lijkt een bijvoegelijk naamwoord voor "company". Maar "gastbedrijven", dat bedoelen we niet. Idem "adminbedrijven". Vanuit deltaportaal meen ik me een `profile.companies_i_manage` te herinneren. Zoiets. Oh wacht, het is TRS. `person.projects_i_manage.all()`

`companies_as_guest`? Die is weer wat minder.

=> Even over brainstormen